aggregate_root/spec/snapshot_repository_spec.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+module AggregateRoot
+  class ReadingStatsRepository < SimpleDelegator
+    def initialize(repository)
+      super(repository)
+      @records_read = 0
+    end
+
+    attr_reader :records_read
+
+    def reset_records_read
+      @records_read = 0
+    end
+
+    def read(spec)
+      @records_read += count(spec)
+      super
+    end
+  end
+
+  RSpec.describe SnapshotRepository do
+    let(:event_store) do
+      RubyEventStore::Client.new(
+        repository: reporting_repository,
+        mapper: RubyEventStore::Mappers::NullMapper.new
+      )
+    end
+    let(:reporting_repository) { ReadingStatsRepository.new(RubyEventStore::InMemoryRepository.new) }
+    let(:uuid) { SecureRandom.uuid }
+    let(:stream_name) { "Order$#{uuid}" }
+    let(:repository) { AggregateRoot::SnapshotRepository.new(event_store) }
+    let(:order_created) { Orders::Events::OrderCreated.new }
+    let(:order_canceled) { Orders::Events::OrderCanceled.new }
+    let(:order_expired) { Orders::Events::OrderExpired.new }
+
+    let(:order_klass) do
+      class Order
+        include AggregateRoot
+
+        def initialize(uuid)
+          @status = :draft
+          @uuid = uuid
+        end
+
+        attr_accessor :status
+
+        def apply_order_created(_)
```

Comment:
`on` methods are preferred since we've scheduled dropping `apply_event_type` meta